pubs/collab-patterns/collab.Rmd

```diff
@@ -26,9 +26,173 @@ output:
 ```{r code = readLines("../load_data.R"), echo = F, message = F}
 ```
 
+Rewritting some of the shared_utils functions to be used in R.
+
+```{r}
+author_name <- function(name) {
+  
+  m <- str_match(name, "([^\\(]*) \\((.*)\\)$")
+  if (is.na(m[,1])) {
+    
+    orig <- name
+    affil <- NA
+    
+    
+  } else {
+    
+    orig <- m[,2]
+    affil <- m[,3]
+    
+  }
+  
+  return(c(orig, affil))
+  
+}
+
+normalized_author_name <- function(name) {
+  
+  recased <- str_to_title(name)
+  
+  recased <- str_replace(recased, "Jr\\.", "") %>% 
+    str_replace("Sr\\.", "") %>% 
+    str_replace("Dr\\.", "") %>% 
+    str_replace("Prof\\.", "")
+  
+  names <- author_name(recased)[1]
+  names <- str_split(names, " ")
+  
+  names <- names[[1]]
+  
+  if (length(names) == 1) {
+
+    return(names)
+
+  } else {
+
+    last <- tail(names, n = 1)
+    rest <- names[1:(length(names) - 1)]
+    
+    if (length(rest) == 1) {
+      first <- rest
+    } else {
+      
+      tot <- ""
+      for (i in seq_along(rest)) {
+        tot <- paste0(tot,' ', rest[i])
+      }
+    
+      first <- tot
+      
+    }
+    
+    full <- paste0(last, ", ", first)
+    return(full)
+
+  }
+  
+}
+
+```
+
+
 
 ```{r setup, echo=F, message=F, warning=F, cache=F}
 library('tidyverse')
+library('rjson')
+
+inferred <- read_csv("https://raw.githubusercontent.com/eitanf/sysconf/master/data/nonsys_inferred_gender_mapping.csv")
```

Comment:
These are not ideal variable names, because we already have `inferred_gender` and `verified_gender` in `load_data.R`. How about "inferred_gender_nonsys", etc?